openeo/rest/auth/testing.py

```diff
@@ -83,6 +85,12 @@ def __init__(
             self.requests_mock.post(
                 self.device_code_endpoint, text=self.device_code_callback
             )
+            if self.use_verification_uri_complete:
+                self._set_user_code()
+                end_point = (
+                    f"{self.device_code_endpoint}?user_code={self.state['user_code']}"
+                )
+                self.requests_mock.post(end_point, text=self.device_code_callback)
```

Comment:
This should be a GET mock instead of a POST mock (the user just clicks this link).
This is actually also wrong on line 85 above

Reply:
sorry I'm confused myself:
- `self.device_code_endpoint` is actually a POST,
- there is no need for a mock of `f"{self.device_code_endpoint}?user_code={self.state['user_code']}"`

Reply:
N.B. I will continue on your branch, just documenting the resolution of the comment here.

Yes, I was not sure whether this extra mock was needed to test the URL that included the query paramater, but we don't need it.

Removing this mock also makes the rest of the code a lot simpler, because those ugly changes that generate the user code at construction time were only necessary to have the user code at the time we set up the mock.